sympy/functions/elementary/complexes.py

```diff
@@ -471,15 +472,26 @@ def eval(cls, arg):
             raise TypeError("Bad argument type for Abs(): %s" % type(arg))
         # handle what we can
         arg = signsimp(arg, evaluate=False)
+        n, d = arg.as_numer_denom()
+        if d.free_symbols and not n.free_symbols:
+            return cls(n)/cls(d)
+
         if arg.is_Mul:
             known = []
             unk = []
             for t in arg.args:
-                tnew = cls(t)
-                if isinstance(tnew, cls):
-                    unk.append(tnew.args[0])
+                if t.is_Pow and t.exp.is_integer and t.exp.is_negative:
+                    bnew = cls(t.base)
+                    if isinstance(bnew, cls):
+                        unk.append(t)
+                    else:
+                        known.append(bnew**t.exp)
```

Comment:
Or:
```suggestion
                        known.append(Pow(bnew, t.exp))
```
Not sure if there's an advantage to skip the Python handling?